packages/yswaps/scripts/libraries/solvers/MulticallDexes.ts

```diff
@@ -0,0 +1,145 @@
+
+    console.log('[Dexes] Total balance is', utils.formatUnits(amount, inDecimals), inSymbol);
+    console.log('[Dexes] Getting', inSymbol, '=>', outSymbol, 'trade information');
+
+    const network = process.env.HARDHAT_DEPLOY_FORK as SUPPORTED_NETWORKS_MOCK;
+    const dexesMap = dexesNerworkMapMock[network];
+    const dexes = Object.values(dexesMap);
+    const dexesBestResults: DexLibrarySwapResponse[] = []; // index will represent the path order.
+
+    // PAIR-IN: tokenIn + hopToken
+    // For optimization we first create an array of all the promises for the first path requests.
+    const firstPathPromises = hopTokens.map((hopToken) => {
+      return dexes.map((dex) => {
+        return dex.swap({
+          tokenIn: tokenInAddress,
+          tokenOut: hopToken,
+          amountIn: amount,
+          strategy,
+        });
+      })
+    }).flat();
+
+    // we call all the promises at onces.
+    let firstBestResponse: DexLibrarySwapResponse;
+    const firstPathResponses = await Promise.all(firstPathPromises);
+    firstPathResponses.forEach((response) => {
+      if (!firstBestResponse || response.amountOut.gt(firstBestResponse.amountOut)) {
+        dexesBestResults[0] = response;
+      };
+    });
+
+    // PAIR-OUT: hopToken + tokenOut
+    // For optimization we first create an array of all the promises for the last path requests.
+    const lastPathPromises = hopTokens.map((hopToken) => {
+      return dexes.map((dex) => {
+        return dex.swap({
+          tokenIn: tokenInAddress,
+          tokenOut: hopToken,
+          amountIn: amount,
+          strategy,
+        });
+      })
+    }).flat();
+
+    // we call all the promises at onces.
+    let lastBestResponse: DexLibrarySwapResponse;
+    const lastPathResponses = await Promise.all(lastPathPromises);
+    lastPathResponses.forEach((response) => {
+      if (!lastBestResponse || response.amountOut.gt(lastBestResponse.amountOut)) {
+        dexesBestResults[0] = response;
+      };
+    });
+
+    const firstSwapResponse = dexesBestResults[0];
+    const lastSwapResponse = dexesBestResults[1];
+
+    // TODO: check if both paths uses same dex.
+    let uniqueSwapResponse: DexLibrarySwapResponse | undefined;
+    if (firstSwapResponse.dex === lastSwapResponse.dex) {
+      // NOTE first draft, could change.
+      // const dex = dexesMap[firstSwapResponse.dex];
+      // uniqueSwapResponse = await dex.swap({
+      //     tokenIn: tokenInAddress,
+      //     tokenOut: tokenOutAddress,
+      //     amountIn: amount,
+      //     strategy,
+      //   });
+    }
+
+    const amountOut = uniqueSwapResponse ? uniqueSwapResponse.amountOut: lastSwapResponse.amountOut;
+    const minAmountOut = amountOut.sub(amountOut.mul(3).div(100));
+    const data = uniqueSwapResponse
+      ? mergeTransactions([uniqueSwapResponse.unsignedSwapTx]) // TODO check if theres a util for single tx
+      : mergeTransactions(dexesBestResults.map((result => result.unsignedSwapTx)));
```

Comment:
This is not exactly right. Since we would need to add check and approvals for each "path". Example: A => H => B. Not necesarry to be applied on this branch. But this code like this, won't work on production.